### PR TITLE
map repository to project using slack modal submission

### PR DESCRIPTION
**Pull request by zach-bob**
This PR implements [this ticket](https://mojotech.atlassian.net/jira/software/projects/MRNN/boards/22?selectedIssue=MRNN-40), which is the last part needed for listening to release note events in slack.

### CODE REVIEW

**Comment by zach-bob** [comment]
Adding DO NOT MERGE due to all of the dependent branches that have not been merged yet.

File: lib/release_notes_bot_web/controllers/webhook_controller.ex [hunk 1]
```diff
@@ -29,7 +29,7 @@ defmodule ReleaseNotesBotWeb.WebhookController do
 
       _ ->
         # Check if repo exists.
-        {:ok, repo} = Repositories.find_or_create(body["repository"])
+        {:ok, repo} = Repositories.webhook_find_or_create(body["repository"])
```
Comment (sethkrasnianski) [inline lib/release_notes_bot_web/controllers/webhook_controller.ex:32]: What about `find_or_create_by_{thing}` as a pattern? `find_or_create_by_webhook` in this case

File: lib/release_notes_bot_web/controllers/slack_interaction_controller.ex [hunk 2]
```diff
@@ -16,10 +16,10 @@ defmodule ReleaseNotesBotWeb.SlackInteractionController do
           "#{user["name"]} has configured this channel to accept messages and updates for projects under: #{client_name}"
         )
 
-      %{client: client_name, project: project_name} ->
-        Channels.post_message(
-          Application.get_env(:release_notes_bot, :slack_channel),
-          "#{user["name"]} has created a new project for #{client_name} titled: '#{project_name}'"
+      %{client: client, project: project_name} ->
+        Channels.post_message_all_client_channels(
```
Comment (sethkrasnianski) [inline lib/release_notes_bot_web/controllers/slack_interaction_controller.ex:20]: This seems unrelated to `add repository url field to slack project modal `. This commit as a whole seems to be doing quite a few different things

Comment (zach-bob) [reply lib/release_notes_bot_web/controllers/slack_interaction_controller.ex:20]: By adding the repo url field, it changed how the modal is parsed, and what is returned when it is parsed. When a new project modal is parsed, it now returns a client and its associate channels. Instead of pulling the slack channel from env, we now are able to take those channels and can broadcast a message to all client channels. I could break all this stuff up or rename the commit message to make everything more clear.

File: lib/release_notes_bot_web/controllers/webhook_controller.ex (hunk 1 above)
Comment (zach-bob) [reply lib/release_notes_bot_web/controllers/webhook_controller.ex:32]: This seems like a great idea
